appserver/distributions/glassfish-common/src/main/resources/bin/stopserv.bat

```diff
@@ -16,4 +17,21 @@ REM  SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
 REM
 
 
-java -jar "%~dp0..\glassfish\modules\admin-cli.jar" stop-domain %*
+VERIFY OTHER 2>nul
+setlocal ENABLEEXTENSIONS
+if ERRORLEVEL 0 goto ok
+echo "Unable to enable extensions"
+exit /B 1
+
+:ok
+set AS_INSTALL=%~dp0..
+call "%AS_INSTALL%\config\asenv.bat"
+if "%AS_JAVA%x" == "x" goto UsePath
+set JAVA=%AS_JAVA%\bin\java

```

Comment:
I guess this should be in quotes in case it contains spaces:
```suggestion
set JAVA="%AS_JAVA%"\bin\java

```

Reply:
As I wrote in another comment, the correct solution is `set "JAVA=%AS_JAVA%\bin\java"`